tests/tsconfig.tests.json

```diff
@@ -1,26 +1,27 @@
 {
   "compilerOptions": {
-      "target": "es2016",
-      "module": "ES2015",
-      "noImplicitAny": true,
-      "sourceMap": false,
-      "downlevelIteration": true,
-      "noUnusedLocals": false,
-      "noUnusedParameters": false,
-      "strictNullChecks": true,
-      "strictFunctionTypes": true,
-      "declaration": true,
-      "moduleResolution": "Node",
-      "baseUrl": "./src",
-      "lib": [
-        "dom",
-        "es2016",
-        "es2017.object"
-      ],
-      "outDir": "dist",
+    "target": "es2016",
+    "module": "ES2015",
+    "noImplicitAny": true,
+    "sourceMap": false,
+    "downlevelIteration": true,
+    "noUnusedLocals": false,
+    "noUnusedParameters": false,
+    "strictNullChecks": true,
+    "strictFunctionTypes": true,
+    "declaration": true,
+    "skipLibCheck": true,
```

Comment:
Only change here is adding skipLibCheck: true to remove an erroneous error when I upgraded

Reply:
I was curious to know what the issues were. Here they are (at least for posterity): 

<img width="1387" alt="image" src="https://github.com/user-attachments/assets/d6dd77d7-8850-4ac2-9b64-d021491f1af5">

... interestingly `WeakKey` is infact a ts builtin:

<img width="874" alt="image" src="https://github.com/user-attachments/assets/6e288106-bc88-4817-a31e-c05254382303">

As for the "Cannot find module 'rollup/parseAst' or its corresponding type declarations." issue, I came across this thread:

https://github.com/rollup/rollup/issues/5199

Any thoughts @dmmulroy

Reply:
https://github.com/vitest-dev/vitest/issues/4921

`WeakKey` is available in TS >= 5.2

Reply:
Errors thrown in `node_modules` don't matter - it's not code that we control. If there are no errors on install (i.e. peerDeps are all good) then we're happy.